src/Store.cs

```diff
@@ -162,6 +164,9 @@ public Store(Engine engine, object? data)
         /// </summary>
         public void GC()
         {
+            _funcCache.Clear();
```

Comment:
Yeah I wasn't too sure about this either. I wanted to have _some_ way to preventing infinite growth of the `Store` and `GC()` seemed like the best existing option, but it's not exactly a perfect fit.

You're correct about things living forever in the `Store`, to quote the Wasmtime docs:

> A [Store](https://docs.wasmtime.dev/api/wasmtime/struct.Store.html) is intended to be a short-lived object in a program. No form of GC is implemented at this time so once an instance is created within a [Store](https://docs.wasmtime.dev/api/wasmtime/struct.Store.html) it will not be deallocated until the [Store](https://docs.wasmtime.dev/api/wasmtime/struct.Store.html) itself is dropped. This makes [Store](https://docs.wasmtime.dev/api/wasmtime/struct.Store.html) unsuitable for creating an unbounded number of instances in it because [Store](https://docs.wasmtime.dev/api/wasmtime/struct.Store.html) will never release this memory. It’s recommended to have a [Store](https://docs.wasmtime.dev/api/wasmtime/struct.Store.html) correspond roughly to the lifetime of a “main instance” that an embedding is interested in executing.

Given this I've removed the cache clearing for now. We can look into more cache control in the future if necessary.